src/QueryReflection/QuerySimulation.php

```diff
@@ -72,13 +75,21 @@ public static function simulateParamValueType(Type $paramType, bool $preparedPar
         }
 
         $stringType = new StringType();
-        if ($stringType->isSuperTypeOf($paramType)->yes() || $paramType instanceof ObjectType && $paramType->isInstanceOf(Stringable::class)->yes()) {
+        $isStringableObjectType = $paramType instanceof ObjectType
+            && $paramType->isInstanceOf(Stringable::class)->yes();
+        $isDateTimeType = $paramType instanceof ObjectType
+            && $paramType->isInstanceOf(DateTimeInterface::class)->yes();
+        if (
+            $stringType->isSuperTypeOf($paramType)->yes()
+            || $isStringableObjectType
+            || $isDateTimeType
+        ) {
             // in a prepared context, regular strings are fine
             if (true === $preparedParam) {
                 // returns a string in date-format, so in case the simulated value is used against a date/datetime column
                 // we won't run into a sql error.
                 // XX in case we would have a proper sql parser, we could feed schema-type-dependent default values in case of strings.
-                return date(MysqliQueryReflector::DATE_FORMAT, 0);
+                return $isDateTimeType ? date(self::DATE_FORMAT, 0) : '1';
```

Comment:
moving the DATE_FORMAT into the simulation would only make sense if it works also for PGSQL. does it?